spike_main/spike.cc

```diff
@@ -189,7 +189,20 @@ static std::vector<mem_cfg_t> parse_mem_layout(const char* arg)
       exit(EXIT_FAILURE);
     }
 
-    res.push_back(mem_cfg_t(reg_t(base), reg_t(size)));
+    mem_cfg_t mem_region(base, size);
+    const uint64_t max_allowed_pa = (1ull << MAX_PADDR_BITS) - 1ull;
+    if ((mem_region.base + mem_region.size - 1) > max_allowed_pa) {
+      fprintf(stderr, "unsupported memory region "
+                      "{ base = 0x%llX, size = 0x%llX } specified\n"
```

Comment:
[Use `cinttypes`](https://stackoverflow.com/a/9225648/202201) for printing `uint64_t`: `PRIX64`

Or C++ streams to `cerr`.

Reply:
Thanks! Decided to use PRIX64 since we tend to use fprintf in other places.

Reply:
One thing to note - to avoid compiler warnings I had to update the surrounding code a little bit (auto -> explicit uint64_t and started to use PRIx64 in another error message). Hope this is not  a big deal.